File: trunk/ShadowFrameworkLite20/src/shadow/pipeline/java/SFGL20ValuenfArray.java
package shadow.pipeline.java;

import shadow.math.SFValuenf;
import shadow.system.SFArrayElementException;

public class SFGL20ValuenfArray extends SFGL20ListData<SFValuenf>{

	private int n;
	
	public SFGL20ValuenfArray(int n) {
		super();
		this.n = n;
	}

	@Override
	protected void assignValues(SFValuenf writing, SFValuenf reading)
			throws SFArrayElementException {
		writing.set(reading);
	}
	
	@Override
	protected SFValuenf generateGenericElement() {
		return new SFValuenf(n);
	}

}

File: ShadowFrameworkJS/src/shadow/pipeline/openGL20/SFGL20ValuenfArray.js
function SFGL20ValuenfArray(n){
	this.data=new Array();
	this.n=n;
}

inherit(SFGL20ValuenfArray,SFGL20ListData);


SFGL20ValuenfArray.prototype["assignValues"]=function(writing,reading){
	writing.setValue(reading);
};

SFGL20ValuenfArray.prototype["generateGenericElement"]=function(){
		return new SFValuenf(this.n);
};
